### DISCUSSION TITLE
Write-DbaDataTable error

### DISCUSSION THREAD

**Discussion by thejosh52** [dataplat/dbatools#7814] [General]
New-Object : Cannot find an overload for "SqlBulkCopy" and the argument count: "3".
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.21\allcommands.ps1:82878 char:21
+ ... $bulkCopy = New-Object Microsoft.Data.SqlClient.SqlBulkCopy($server.C ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-Object], MethodException
    + FullyQualifiedErrorId : ConstructorInvokedThrowException,Microsoft.PowerShell.Commands.NewObjectCommand

The property 'DestinationTableName' cannot be found on this object. Verify that the property exists and can be set.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.21\allcommands.ps1:82880 char:9
+         $bulkCopy.DestinationTableName = $fqtn
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'BatchSize' cannot be found on this object. Verify that the property exists and can be set.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.21\allcommands.ps1:82881 char:9
+         $bulkCopy.BatchSize = $BatchSize
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'NotifyAfter' cannot be found on this object. Verify that the property exists and can be set.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.21\allcommands.ps1:82882 char:9
+         $bulkCopy.NotifyAfter = $NotifyAfter
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

The property 'BulkCopyTimeOut' cannot be found on this object. Verify that the property exists and can be set.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.21\allcommands.ps1:82883 char:9
+         $bulkCopy.BulkCopyTimeOut = $BulkCopyTimeOut
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : PropertyNotFound

You cannot call a method on a null-valued expression.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.21\allcommands.ps1:82893 char:9
+         $bulkCopy.Add_SqlRowsCopied( {
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

Name                           Value
----                           -----
PSVersion                      5.1.17763.2183
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.2183
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

**Comment by andreasjordan** [comment]
Most likely a problem with an also imported SqlServer module. Can you check that? Always import dbatools first.
